src/views/common/TimeScrollView/YAxisTicks.tsx

```diff
@@ -32,27 +32,65 @@ const range = (min: number, max: number, step: number, base: number) => {
+        const realizedScale = Math.pow(10, scale)   // realizedScale = the base order of magnitude represented by `scale`.
         const results: {step: number, scale: number}[] = []
-        for (let s of steps) {
+        for (let s of steps) {  // for each of the 3 candidate step sizes at this order of magnitude
+            // determine how many steps of [1 | 2 | 5] * 10^scale would be needed to exhaust the range.
+            // (technically on the plot this will be the number of spaces between the grid lines, thus the hard </>).
             const fit = range/(s * realizedScale)
             if (fit > minGridLines && fit < maxGridLines) {
+                // if this split produces an acceptable number of counts, record the step and scale sizes that work.
                 results.push({step: s, scale: scale})
             }
-            // this means the step size is too big. This shouldn't really happen without finding an acceptable step size first,
-            // but we'll check for it later just in case.
-            if (fit < minGridLines) { return {step: -1, scale: -1} }
-            results.push({step: 0, scale: 0})
+            if (fit < minGridLines) {
+                // step size was too big--it resulted in too few divisions of the range.
+                results.push({step: -1, scale: -1})
+                break
+            }
         }
+        // Completing this loop means we finished evaluating one order of magnitude. If we found
+        // a step-scale combination that gave a number of divisions in the acceptable range,
+        // it should show up as a result record with a positive step size (...viz, 1, 2, or 5).
+        // We never have to worry about it actually being from a different scale, because we run
+        // this operation every time we complete an order of magnitude.
         const a = results.find(r => r.step > 0)
         if (a) return a
+        // On the other hand, if the result set includes an entry with a step size of -1, then
+        // the last loop's step size got too big. This is bad--if we see this, it means it
+        // is not productive to try larger step sizes, but if we had found something that did
+        // work, we would've returned that first. So realistically this probably means the
+        // range of allowed grid lines is too small. (Those are set programmatically from
+        // a range of allowable inter-tick pixel counts, minGridSpacingPx and maxGridSpacingPx,
+        // which might need to be adjusted.)
         const b = results.find(r => r.step === -1)
         if (b) return b
-        scale = scale + 1
+        scale = scale + 1 // Try the next larger order of magnitude.
+        // If a step size of 50_000_000_000 base units still produces too many lines, give up.
+        // (This should never happen, since we normalized the input range to 100-1000.)
         if (scale > 10) {
             return {step: 1, scale: 0}
```

Comment:
I vote for returning undefined here. We could not find a proper scale. The calling code can then figure out what to do with this. I don't like signalling by returning `{step: 1, scale: 0}`.
Also, if we handled negative values of scale, then we would never have to return the other signal `{step: -1, scale: -1}`.

Reply:
This isn't signaling--it's just defaulting to a unit stepsize.

Reply:
ah, okay.